src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/ProcessExecutor.cs

```diff
@@ -0,0 +1,186 @@
+            string command,
+            string arguments,
+            string? workingDir,
+            CancellationToken cancellationToken,
+            TimeSpan? timeout = null)
+        {
+            if (string.IsNullOrWhiteSpace(command))
+            {
+                Logger.LogError("Security violation: Attempted to execute null or empty command");
+                throw new ArgumentException("Command cannot be null or empty", nameof(command));
+            }
+
+            if (!SecureProcessConfiguration.IsCommandAllowed(command))
+            {
+                Logger.LogError("Security violation: Attempted to execute unauthorized command: {Command}", command);
+                throw new UnauthorizedAccessException($"Command '{command}' is not in the allowed commands list");
+            }
+
+            if (!string.IsNullOrEmpty(workingDir))
+            {
+                ValidationResult workingDirValidation = InputValidator.ValidateWorkingDirectory(workingDir);
+                if (!workingDirValidation.IsValid)
+                {
+                    Logger.LogError("Security violation: Invalid working directory: {WorkingDirectory} - {Error}", 
+                        workingDir, workingDirValidation.ErrorMessage);
+                    throw new UnauthorizedAccessException($"Working directory '{workingDir}' failed security validation: {workingDirValidation.ErrorMessage}");
+                }
+            }
+
+            arguments ??= string.Empty;
+
+            Logger.LogInformation("Executing validated command: {Command} with args: {Arguments} in directory: {WorkingDirectory}", 
+                command, 
+                arguments.Length > 100 ? arguments.Substring(0, 100) + "..." : arguments,
+                workingDir ?? Environment.CurrentDirectory);
+
+            try
+            {
+                using Process process = CreateProcess(command, arguments, workingDir);
+
+                process.Start();
+
+                Task<string> outputTask = process.StandardOutput.ReadToEndAsync();
+                Task<string> errorTask = process.StandardError.ReadToEndAsync();
+
+                if (timeout.HasValue)
+                {
+                    using CancellationTokenSource cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                    cts.CancelAfter(timeout.Value);
+
+                    try
+                    {
+                        await process.WaitForExitAsync(cts.Token).ConfigureAwait(false);
+                    }
+                    catch (OperationCanceledException) when (cts.Token.IsCancellationRequested && !cancellationToken.IsCancellationRequested)
+                    {
+                        Logger.LogError("Process timed out after {Timeout}ms: {Command}", timeout.Value.TotalMilliseconds, command);
+
+                        try
+                        {
+                            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
May be helpful to have a comment here as to why we're doing a platform sniff.   I think you could rewrite to simplify

```csharp
process.Kill(entireProcessTree: RuntimeInformation.IsOSPlatform(OSPlatform.Windows));
```